lib/serializer.ts

```diff
@@ -414,44 +413,47 @@ function serializeDictionaryType(serializer: Serializer, mapper: DictionaryMappe
   return tempDictionary;
 }
 
+/**
+ * Resolves a composite mapper's modelProperties.
+ * @param serializer the serializer containing the entire set of mappers
+ * @param mapper the composite mapper to resolve
+ */
+function resolveModelProperties(serializer: Serializer, mapper: CompositeMapper, objectName: string): { [propertyName: string]: Mapper } {
+  let modelProps = mapper.type.modelProperties;
+  if (!modelProps) {
+    const className = mapper.type.className;
+    if (!className) {
+      throw new Error(`Class name for model "${objectName}" is not provided in the mapper "${JSON.stringify(mapper, undefined, 2)}".`);
+    }
+
+    const modelMapper = serializer.modelMappers[className];
+    if (!modelMapper) {
+      throw new Error(`mapper() cannot be null or undefined for model "${className}".`);
+    }
+    modelProps = modelMapper.type.modelProperties;
+    if (!modelProps) {
+      throw new Error(`modelProperties cannot be null or undefined in the ` +
+        `mapper "${JSON.stringify(modelMapper)}" of type "${className}" for object "${objectName}".`);
+    }
```

Comment:
Does it make sense to assign `mapper.type.modelProperties = modelProps;` at this point so that we don't have to do the same checks the next time this method is called with that mapper?